src/test/libcephfs/snapdiff.cc

```diff
@@ -497,71 +550,112 @@ TEST(LibCephFS, SnapDiffLib)
+    }
+
+    //
+    // Make sure snap1 vs. snap2 delta for /dirD is as expected
+    //
+    {
+      vector<pair<string, uint64_t>> expected;
+      test_mount.verify_snap_diff(expected, "dirD", "snap1", "snap2");
+    }
+
+    // Make sure SnapDiff returns an error when provided with the same
+    // snapshot name for both parties A and B.
+    {
+      string snap_path = test_mount.make_snap_path("snap2");
+      string snap_other_path = snap_path;
+      std::cout << "---------invalid snapdiff params, the same snaps---------" << std::endl;
+      ASSERT_EQ(-EINVAL, test_mount.for_each_readdir_snapdiff(
+        "",
+        "snap2",
+        "snap2",
+        [&](const dirent* dire, uint64_t snapid) {
+          return true;
+        }));
+    }
+    // Make sure SnapDiff returns an error when provided with an empty
+    // snapshot name for one of the parties
+    {
+      std::cout << "---------invalid snapdiff params, no snap_other ---------" << std::endl;
+      string snap_path = test_mount.make_snap_path("snap2");
+      string snap_other_path;
+      ASSERT_EQ(-EINVAL, test_mount.for_each_readdir_snapdiff(
+        "",
+        "snap2",
+        "",
+        [&](const dirent* dire, uint64_t snapid) {
+          return true;
+        }));
+    }
   }
 
-  // Make sure SnapDiff returns an error when provided with the same
-  // snapshot name for both parties A and B.
-  {
-    string snap_path = test_mount.make_snap_path("snap2");
-    string snap_other_path = snap_path;
-    std::cout << "---------invalid snapdiff params, the same snaps---------" << std::endl;
-    ASSERT_EQ(-EINVAL, test_mount.for_each_readdir_snapdiff(
-      "",
-      "snap2",
-      "snap2",
-      [&](const dirent* dire, uint64_t snapid) {
-        return true;
-      }));
+  json_spirit::mValue dump = test_mount.tell_rank0("dump_historic_ops");
+  ASSERT_FALSE(dump.is_null());
+
+  bool seen_cap_throttle_in_recent_op_events = false;
+  try {
+    for (const auto& op: dump.get_obj().at("ops").get_array()) {
+      for (const auto& ev: op.get_obj().at("type_data").get_obj().at("events").get_array()) {
+        if (ev.get_obj().at("event") == "cap_acquisition_throttle") {
+	  seen_cap_throttle_in_recent_op_events = true;
+	  goto done;
+	}
+      }
+    }
+    done:;
   }
-  // Make sure SnapDiff returns an error when provided with an empty
-  // snapshot name for one of the parties
-  {
-    std::cout << "---------invalid snapdiff params, no snap_other ---------" << std::endl;
-    string snap_path = test_mount.make_snap_path("snap2");
-    string snap_other_path;
-    ASSERT_EQ(-EINVAL, test_mount.for_each_readdir_snapdiff(
-      "",
-      "snap2",
-      "",
-      [&](const dirent* dire, uint64_t snapid) {
-        return true;
-      }));
+  catch (const std::runtime_error &e) {
+    std::cout << "error while parsing dump_historic_ops: " << e.what() << std::endl;
   }
 
+  ASSERT_TRUE(seen_cap_throttle_in_recent_op_events);
```

Comment:
@leonid-s-usov PTAL test failures here - https://pulpito.ceph.com/vshankar-2023-08-09_05:51:38-fs-wip-vshankar-testing-20230808.093601-testing-default-smithi/7364204/

```
2023-08-09T06:50:26.512 INFO:tasks.workunit.client.0.smithi002.stdout:/build/ceph-18.0.0-5396-ge318c197/src/test/libcephfs/snapdiff.cc:655: Failure
2023-08-09T06:50:26.513 INFO:tasks.workunit.client.0.smithi002.stdout:Value of: seen_cap_throttle_in_recent_op_events
2023-08-09T06:50:26.514 INFO:tasks.workunit.client.0.smithi002.stdout:  Actual: false
2023-08-09T06:50:26.514 INFO:tasks.workunit.client.0.smithi002.stdout:Expected: true
```

The throttle didn't get hit and caused the assert. I haven't looked at the MDS logs to infer why. In case you don't have access to sepia yet, the (MDS) logs can be accessed via: http://qa-proxy.ceph.com/teuthology/vshankar-2023-08-09_05:51:38-fs-wip-vshankar-testing-20230808.093601-testing-default-smithi/7364204/remote/

Reply:
found the mds logs, we can see the evidence also there: 
```
2023-08-09T06:50:24.477+0000 7fe25c929700 20 mds.0.server snapdiff throttled. max_caps_per_client: 1 num_caps: 11 session_cap_acquistion: 1.98895 cap_acquisition_throttle: 1
```

Reply:
> Is this test running against a shared ceph cluster? Could it be that the historic ops are overflown by the time we're issuing the `dump_historic_ops` command?

```
# Max number of completed ops to track
- name: mds_op_history_size
  type: uint
  level: advanced
  desc: maximum size for list of historical operations
  default: 20
  services:
  - mds
  with_legacy: true
```

You are probably correct.